src/openloco/map/tile.cpp

```diff
@@ -189,4 +196,56 @@ namespace openloco::map
 
         return coordinate2D;
     }
+
+    xy32 mapWindowPosToLocation(map_pos pos)
+    {
+        pos.x = ((pos.x + 8) - map_columns) / 2;
+        pos.y = ((pos.y + 8)) / 2;
+        xy32 location = { static_cast<coord_t>(pos.y - pos.x), static_cast<coord_t>(pos.x + pos.y) };
+        location.x *= tile_size;
+        location.y *= tile_size;
+
+        switch (getCurrentRotation())
+        {
+            case 0:
+                return location;
+            case 1:
+                return { map_width - 1 - location.y, location.x };
+            case 2:
+                return { map_width - 1 - location.x, map_height - 1 - location.y };
+            case 3:
+                return { location.y, map_height - 1 - location.x };
+        }
+
+        return { 0, 0 }; // unreachable
+    }
+
+    map_pos locationToMapWindowPos(xy32 pos)
```

Comment:
You have the types back to front. This takes a `map_pos` and returns an `xy32`. Look at all of the callers they are all things that have x, y coordinates.

Reply:
`map_pos` is the position on the game map not the position on the map window map